lib/serializer.js

```diff
@@ -157,8 +161,17 @@ function _getSerializedContext(err) {
         _.omit(err, self._knownFields) :
         {};
 
-    if (topLevelFields.domain instanceof domain.Domain) {
```

Comment:
This may not be necessary anymore [now that Node.js sets the `domain` property on objects as being non-enumerable](https://github.com/nodejs/node/commit/377c5835e8bfcd04273f24a244b2ba4aff76a27c) for versions of node >= 12.0.0

Reply:
I guess we still need to support Node.js >= 10.x, and possibly older versions (like 8.x as indicated by the Travis config) though.

Reply:
I guess we should keep supporting all Node.js versions which are still active (8.x, 10.x, 12.x and 13.x)? If that's the case, it will take some time for us to leverage the changes introduced on v12 (out of curiosity, those changes are also part of the performance issues I found).

Reply:
Although 8.x goes EOL later this month, so we can drop that soon...